modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/EncryptionUtil.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.persistence.file;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import org.apache.ignite.internal.managers.encryption.GroupKey;
+import org.apache.ignite.internal.processors.cache.persistence.tree.io.PageIO;
+import org.apache.ignite.internal.processors.cache.persistence.wal.crc.FastCrc;
+import org.apache.ignite.internal.util.typedef.internal.CU;
+import org.apache.ignite.spi.encryption.EncryptionSpi;
+
+/** Encryption utililty class. */
+public class EncryptionUtil {
+    /** Encryption SPI. */
+    private final EncryptionSpi encSpi;
+
+    /** Page size. */
+    private final int pageSize;
+
+    /** Extra bytes added by encryption. */
+    private final int encryptionOverhead;
+
+    /**
+     * Array of zeroes to fulfill tail of decrypted page.
+     */
+    private final byte[] zeroes;
+
+    /**
+     * @param spi Spi.
+     * @param pageSize Page size.
+     */
+    public EncryptionUtil(EncryptionSpi spi, int pageSize) {
+        encSpi = spi;
+        this.pageSize = pageSize;
+
+        this.encryptionOverhead = pageSize - CU.encryptedPageSize(pageSize, encSpi);
+        this.zeroes = new byte[encryptionOverhead];
+    }
+
+    /**
+     * @param srcBuf Source buffer.
+     * @param res Destination buffer.
+     * @param grpKey Encryption key.
+     */
+    public void encrypt(ByteBuffer srcBuf, ByteBuffer res, GroupKey grpKey) {
+        assert srcBuf.remaining() >= pageSize;
```

Comment:
Please always use the human-readable message with the assertion here and below.
I think it really helps to analyze log files in case of incidents.